### PR TITLE
Checks existence of node before getting SecurityGroups and Images

### PR DESCRIPTION
**Pull request by danielestevez**

### CODE REVIEW

File: azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/extensions/AzureComputeSecurityGroupExtension.java [hunk 1]
```diff
@@ -123,6 +123,9 @@
       ResourceGroup resourceGroup = resourceGroupMap.getUnchecked(regionAndId.region());
 
       VirtualMachine vm = api.getVirtualMachineApi(resourceGroup.name()).get(regionAndId.id());
```
Comment (andreaturli) [inline azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/extensions/AzureComputeSecurityGroupExtension.java]: Can you use checkNotNull, maybe?

File: azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/functions/ResourceDefinitionToCustomImage.java [hunk 2]
```diff
@@ -68,6 +68,9 @@ public Image apply(ResourceDefinition input) {
       ResourceGroup resourceGroup = resourceGroupMap.getUnchecked(regionAndId.region());
       
       VirtualMachine vm = api.getVirtualMachineApi(resourceGroup.name()).get(regionAndId.id());
+      if (vm == null) {
```
Comment (andreaturli) [inline azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/functions/ResourceDefinitionToCustomImage.java]: Same as above

File: azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/extensions/AzureComputeSecurityGroupExtension.java (hunk 1 above)
Comment (nacx) [inline azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/extensions/AzureComputeSecurityGroupExtension.java]: I'd say it is ok as-is? taht check would throw a NPE, but in this case an IAE makes more sense?

File: azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/extensions/AzureComputeImageExtension.java [hunk 3]
```diff
@@ -123,6 +123,8 @@ public Image call() throws Exception {
 
             Image image = resourceDefinitionToImage.create(cloneTemplate.getSourceNodeId(), cloneTemplate.getName())
                   .apply(definitions.get(0));
+            checkState(image != null, "Image %s was not created for the requested node",
+                  cloneTemplate.getSourceNodeId());
```
Comment (nacx) [inline azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/extensions/AzureComputeImageExtension.java]: The log parameter is the id of the node. Better use a message like "Image *for node %s*..."

**Comment by danielestevez** [comment]
:up: Using checkNull may be more readable but i agree it should throw an IAE. Thoughts? @nacx @andreaturli

File: azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/extensions/AzureComputeImageExtension.java (hunk 3 above)
Comment (nacx) [inline azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/extensions/AzureComputeImageExtension.java]: Fix this log message too

File: azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/extensions/AzureComputeImageExtension.java [hunk 4]
```diff
@@ -123,8 +123,7 @@ public Image call() throws Exception {
 
             Image image = resourceDefinitionToImage.create(cloneTemplate.getSourceNodeId(), cloneTemplate.getName())
                   .apply(definitions.get(0));
-            checkState(image != null, "Image %s was not created for the requested node",
-                  cloneTemplate.getSourceNodeId());
+            checkState(image != null, "Image for node %s was not be created", cloneTemplate.getSourceNodeId());
```
Comment (nacx) [inline azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/extensions/AzureComputeImageExtension.java]: "was not **be** created?" :)

Comment (danielestevez) [inline azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/extensions/AzureComputeImageExtension.java]: oops, sorry

**Comment by nacx** [comment]
Squashed and pushed to [master](http://git-wip-us.apache.org/repos/asf/jclouds-labs/commit/550ca7d5) and [2.0.x](http://git-wip-us.apache.org/repos/asf/jclouds-labs/commit/e4e42454). Thanks @danielestevez!

**Pull request closed without merging**